src/test/java/org/elasticsearch/test/discovery/ClusterDiscoveryConfiguration.java

```diff
@@ -126,22 +134,47 @@ public Settings node(int nodeOrdinal) {
                     .put("discovery.zen.ping.multicast.enabled", false);
 
             String[] unicastHosts = new String[unicastHostOrdinals.length];
-            String mode = nodeSettings.get("node.mode", InternalTestCluster.NODE_MODE);
-            if (mode.equals("local")) {
+            if (localMode) {
                 builder.put(LocalTransport.TRANSPORT_LOCAL_ADDRESS, "node_" + nodeOrdinal);
                 for (int i = 0; i < unicastHosts.length; i++) {
                     unicastHosts[i] = "node_" + unicastHostOrdinals[i];
                 }
             } else {
                 // we need to pin the node port & host so we'd know where to point things
-                builder.put("transport.tcp.port", basePort + nodeOrdinal);
+                builder.put("transport.tcp.port", unicastHostPorts[nodeOrdinal]);
```

Comment:
can we check that nodeOrdinal is < unicastHostPorts and throw an exception? I think in theory we don't need to enforce this but can search for ports on the fly, but let's leave that to a day we need it.